### ISSUE TITLE
Add support for Fallback.ofAsync(Supplier<CompletionStage>)

### ISSUE THREAD

**Issue by jhalterman** [jhalterman/failsafe#165]
Add support for supplying CompletionStage to Fallbacks:

```
public static <R> Fallback<R> ofAsync(Supplier<CompletionStage<? extends R>> fallback) ...
```

See also https://github.com/jhalterman/failsafe/issues/145#issuecomment-456164754.

**Comment by georgeteo** [comment]
+1 to this. I implemented `AsyncFallback` policy and policy executor myself to get around this.

**Comment by jhalterman** [comment]
This was released in 2.3.3.
